docs/mivot/viewer.rst

```diff
@@ -23,66 +23,138 @@ Introduction
 Using the API
 =============
 
+ .. attention::
+	The module based on XPath queries and allowing to browse the XML
+	annotations (``viewer.XmlViewer``)  has been removed from version 1.8
+	
 Integrated Readout
 ------------------
 The ``ModelViewer`` module manages access to data mapped to a model through dynamically
 generated objects (``MivotInstance`` class).
 
-The example below shows how a VOTable result of a cone-search query can be parsed and data
-mapped to the ``EpochPosition`` class.
-
-
-.. doctest-remote-data::
-    >>> import astropy.units as u
-    >>> from astropy.coordinates import SkyCoord
-    >>> from pyvo.dal.scs import SCSService
-    >>> from pyvo.utils.prototype import activate_features
-    >>> from pyvo.mivot.version_checker import check_astropy_version
-    >>> from pyvo.mivot.viewer.mivot_viewer import MivotViewer
-    >>> activate_features("MIVOT")
-    >>> if check_astropy_version() is False:
-    ...    pytest.skip("MIVOT test skipped because of the astropy version.")
-    >>> scs_srv = SCSService("https://vizier.cds.unistra.fr/viz-bin/conesearch/V1.5/I/239/hip_main")
-    >>> m_viewer = MivotViewer(
-    ...     scs_srv.search(
-    ...         pos=SkyCoord(ra=52.26708 * u.degree, dec=59.94027 * u.degree, frame='icrs'),
-    ...         radius=0.05
-    ...     ),
-    ...     resolve_ref=True
-    ... )
-    >>> mivot_instance = m_viewer.dm_instance
-    >>> print(mivot_instance.dmtype)
-    mango:EpochPosition
-    >>> print(mivot_instance.spaceSys.frame.spaceRefFrame.value)
-    ICRS
-    >>> while m_viewer.next_row_view():
-    ...     print(f"position: {mivot_instance.latitude.value} {mivot_instance.longitude.value}")
-    position: 59.94033461 52.26722684
-
-
-In this example, the data readout is totally managed by the ``MivotViewer`` instance.
-The ``astropy.io.votable`` API is encapsulated in this module.
-
-Model leaves (class attributes) are complex types that provide additional information:
+These objects can be used as standard Python instances, with the fields representing model elements.
```

Comment:
I assume this are dummy Python instances of the model representation (and not the model itself), i.e. `instance={a:1, b={c=2, type=int}}` is the equivalent of `instance=Object`, `instance.a=1`, `instance.b.c=2`, `instance.b.type='int'`. If that's the case, why bother with both representations and not use only the dictionary?
IMO the instance model, should build actual MANGO instances, enforce attribute types and values, provide convenience methods, etc.

Reply:
I removed any mention if this dual feature in the doc